src/transactions/TransactionUtils.cpp

```diff
@@ -293,12 +295,12 @@ addBuyingLiabilities(LedgerTxnHeader const& header, LedgerTxnEntry& entry,
     }
     else if (entry.current().data.type() == TRUSTLINE)
     {
-        auto& tl = entry.current().data.trustLine();
-        if (!isAuthorized(entry))
+        if (header.current().ledgerVersion < 10 && !isAuthorized(entry))
```

Comment:
same thing here. I'd prefer to just check for `isAuthorizedToMaintainLiabilities` (defensive coding)

Reply:
done